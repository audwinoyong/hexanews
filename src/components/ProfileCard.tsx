import * as React from 'react'

import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'

type Props = {
  name: string,
  description: string,
  image: string,
}

const styles = {
  card: {
    display: 'flex',
    boxShadow: '0px 0px 0px 0px rgba(0, 0, 0, 0.2)',
  },
  details: {
    display: 'flex',
    flex: 4,
    flexDirection: 'column',
  },
  content: {
    flex: '1 0 auto',
  },
  avatar: {
    width: 100,
    height: 100,
    borderRadius: '50%',
  },
}

const ProfileCard: React.StatelessComponent<Props> = ({
  name,
  description,
  image,
}) => {
  return (
    <div>
      <Card style={styles.card as any}>
        <div style={styles.details as any}>
          <CardContent style={styles.content}>
            <Typography variant='display1' component='h1'>
              {name}
            </Typography>
            <Typography component='p' style={{ marginTop: 4 }}>
              {description}
            </Typography>
            <Typography component='p' style={{ marginTop: 12, fontSize: 12, color: 'gray' }}>
              {'Hexanews member since 20 August 2018'}
            </Typography>
          </CardContent>
        </div>
        <div style={{ flex: 1, justifyContent: 'center', alignItems: 'center', display: 'flex' }}>
          <img style={styles.avatar} src={image} alt='avatar' />
        </div>
      </Card>
    </div>
  )
}

export default ProfileCard