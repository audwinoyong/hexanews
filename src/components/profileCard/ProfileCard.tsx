import * as React from 'react'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'

import styles from './styles'

type Props = {
  name: string,
  description: string,
  image: string,
}

const ProfileCard: React.StatelessComponent<Props> = ({
  name,
  description,
  image,
}) => {
  return (
    <div>
      <Card style={styles.card}>
        <div style={styles.details as any}>
          <CardContent style={styles.content}>
            <Typography
              variant='display1'
              component='h1'
            >
              {name}
            </Typography>
            <Typography
              component='p'
              style={styles.description}
            >
              {description}
            </Typography>
            <Typography
              component='p'
              style={styles.memberText}
            >
              {'Hexanews member since 20 August 2018'}
            </Typography>
          </CardContent>
        </div>
        <div style={styles.imageContainer}>
          <img
            style={styles.avatar}
            src={image}
            alt='avatar'
          />
        </div>
      </Card>
    </div>
  )
}

export default ProfileCard