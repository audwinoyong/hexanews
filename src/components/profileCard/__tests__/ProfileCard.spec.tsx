import * as React from 'react'
import * as renderer from 'react-test-renderer'
import * as Enzyme from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

import ProfileCard from '../ProfileCard'

Enzyme.configure({ adapter: new Adapter() })

describe('Profile Card', () => {
  it('renders Profile Card', () => {
    const component = renderer.create(
      <ProfileCard
        name='test'
        description='desc'
        image='test'
        createdAt='asdf'
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})