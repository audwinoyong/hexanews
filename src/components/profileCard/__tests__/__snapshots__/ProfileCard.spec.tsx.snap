// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Card renders Profile Card 1`] = `
<div
  className=""
>
  <div
    className="MuiPaper-root-99 MuiPaper-elevation1-102 MuiPaper-rounded-100 MuiCard-root-98"
    style={
      Object {
        "boxShadow": "0px 0px 0px 0px rgba(0, 0, 0, 0.2)",
        "display": "flex",
      }
    }
  >
    <div
      className=""
      style={
        Object {
          "display": "flex",
          "flex": 4,
          "flexDirection": "column",
        }
      }
    >
      <div
        className="MuiCardContent-root-126"
        style={
          Object {
            "flex": "1 0 auto",
          }
        }
      >
        <h1
          className="MuiTypography-root-127 MuiTypography-display1-131"
        >
          test
        </h1>
        <p
          className="MuiTypography-root-127 MuiTypography-body1-136"
          style={
            Object {
              "marginLeft": 4,
              "marginTop": 4,
            }
          }
        >
          desc
        </p>
        <p
          className="MuiTypography-root-127 MuiTypography-body1-136"
          style={
            Object {
              "color": "gray",
              "fontSize": 12,
              "marginTop": 12,
            }
          }
        >
          Hexanews member since Invalid date
        </p>
      </div>
    </div>
    <div
      className=""
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <div
        className="MuiCardMedia-root-152"
        src={undefined}
        style={
          Object {
            "backgroundImage": "url(\\"test\\")",
            "borderRadius": "50%",
            "height": 100,
            "width": 100,
          }
        }
      />
    </div>
  </div>
</div>
`;
